x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java

```diff
@@ -526,4 +526,8 @@ private static String constantExpression() {
 
     }
 
+    private static String randomIdentifier() {
+        return randomAlphaOfLength(randomIntBetween(8, 12));
```

Comment:
Let's add a comment that explains that having at least 8 chars prevents us from accidentally using the `not` keyword; maybe future us will want the 3 char identifiers back, and will have to exclude `not` specifically.

Reply:
You might borrow/import one from EsTestCase:

https://github.com/elastic/elasticsearch/blob/1943844d5aff294d286f70392b519b7332af7f7a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java#L1278-L1280

Reply:
I'd say let's keep it as it is, the comment is the only real added value, but maybe it will save us from future headaches